pkg/products/threescale/reconciler.go

```diff
@@ -261,7 +276,33 @@ func (r *Reconciler) Reconcile(ctx context.Context, installation *integreatlyv1a
 	return integreatlyv1alpha1.PhaseCompleted, nil
 }
 
-func (r *Reconciler) reconcileTemplates(ctx context.Context, installation *integreatlyv1alpha1.RHMI, serverClient k8sclient.Client) (integreatlyv1alpha1.StatusPhase, error) {
+// restores seed and master api cast secrets if available
+func (r *Reconciler) restoreSystemSecrets(ctx context.Context, serverClient k8sclient.Client, installation *integreatlyv1alpha1.RHMI) (integreatlyv1alpha1.StatusPhase, error) {
+	for _, secretName := range []string{systemSeedSecretName, systemMasterApiCastSecretName} {
+		err := resources.CopySecret(ctx, serverClient, secretName, installation.Namespace, secretName, r.Config.GetNamespace())
+		if err != nil {
+			if !k8serr.IsNotFound(err) {
+				return integreatlyv1alpha1.PhaseFailed, err
+			}
+			logrus.Info(fmt.Sprintf("no backed up secret %v found in %v", secretName, installation.Namespace))
+		}
+	}
+
+	return integreatlyv1alpha1.PhaseCompleted, nil
+}
+
+// Copies the seed and master api cast secrets for later restoration
+func (r *Reconciler) backupSystemSecrets(ctx context.Context, serverClient k8sclient.Client, installation *integreatlyv1alpha1.RHMI) (integreatlyv1alpha1.StatusPhase, error) {
+	for _, secretName := range []string{systemSeedSecretName, systemMasterApiCastSecretName} {
+		err := resources.CopySecret(ctx, serverClient, secretName, r.Config.GetNamespace(), secretName, installation.Namespace)
+		if err != nil {
+			return integreatlyv1alpha1.PhaseFailed, err
```

Comment:
is failed the best state here? The secret could already exist and the error be really quite trivial, right?